### PR TITLE
Label application and version for Kiali observation

### PR DESCRIPTION
**Pull request by davisgomes**
# Description

Labels the application and version for each workload for observation in Kiali.

Fixes VZ-4007

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by anders-swanson — APPROVED** [review]

File: examples/bobs-books/bobs-books-comp.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ spec:
   workload:
     apiVersion: oam.verrazzano.io/v1alpha1
     kind: VerrazzanoCoherenceWorkload
+    metadata:
+      labels:
+        app: robert-coherence
```
Comment (pfmackin) [inline examples/bobs-books/bobs-books-comp.yaml:15]: The label is supposed to be on the pods, does this do that?

Comment (davisgomes) [reply examples/bobs-books/bobs-books-comp.yaml]: Updated the locations of the labels to fulfill this requiremment

File: examples/multicluster/todo-list/todo-list-components.yaml [hunk 2]
```diff
@@ -58,6 +61,9 @@ spec:
     metadata:
       name: tododomain-jdbc-config
       namespace: mc-todo-list
+      labels:
+        app: tododomain-jdbc-config
```
Comment (pfmackin) [inline examples/multicluster/todo-list/todo-list-components.yaml:65]: This is not a workload so i don't think it should be labeled (no pods)

Comment (davisgomes) [reply examples/multicluster/todo-list/todo-list-components.yaml:65]: Okay will update

File: examples/bobs-books/bobs-books-comp.yaml [hunk 3]
```diff
@@ -318,6 +335,7 @@ spec:
           creationTimestamp: null
           labels:
             app: mysql
```
Comment (pfmackin) [inline examples/bobs-books/bobs-books-comp.yaml:337]: app should be bobs-mysql

Comment (davisgomes) [reply examples/bobs-books/bobs-books-comp.yaml:337]: okay, I'll do the same for socks

File: examples/bobs-books/bobs-books-comp.yaml [hunk 4]
```diff
@@ -285,6 +299,9 @@ spec:
     metadata:
       name: mysql
       namespace: bobs-books
+      labels:
+        app: bobs-mysql-service
```
Comment (pfmackin) [inline examples/bobs-books/bobs-books-comp.yaml:303]: change to bobs-mysql

File: examples/bobs-books/bobs-books-comp.yaml [hunk 5]
```diff
@@ -245,6 +256,9 @@ spec:
     metadata:
       name: mysql-initdb-config
       namespace: bobs-books
+      labels:
+        app: bobs-mysql-configmap
```
Comment (pfmackin) [inline examples/bobs-books/bobs-books-comp.yaml:260]: no need to label configmaps

File: examples/bobs-books/bobs-books-comp.yaml [hunk 6]
```diff
@@ -376,6 +394,8 @@ spec:
     metadata:
       labels:
         weblogic.domainUID: bobs-bookstore
+        app: bobs-orders-configmap
```
Comment (pfmackin) [inline examples/bobs-books/bobs-books-comp.yaml:397]: no need to label configmaps

File: examples/multicluster/todo-list/todo-list-components.yaml [hunk 7]
```diff
@@ -99,6 +105,9 @@ spec:
     metadata:
       name: mysql-initdb-config
       namespace: mc-todo-list
+      labels:
+        app: mysql-initdb-config
```
Comment (pfmackin) [inline examples/multicluster/todo-list/todo-list-components.yaml:109]: no need to label configmaps

File: examples/todo-list/todo-list-components.yaml [hunk 8]
```diff
@@ -148,6 +160,7 @@ spec:
         metadata:
           labels:
             app: mysql
```
Comment (pfmackin) [inline examples/todo-list/todo-list-components.yaml:162]: app: todo-mysql

File: examples/todo-list/todo-list-components.yaml [hunk 9]
```diff
@@ -120,6 +129,9 @@ spec:
     metadata:
       name: mysql
       namespace: todo-list
+      labels:
+        app: todo-mysql-service
```
Comment (pfmackin) [inline examples/todo-list/todo-list-components.yaml:133]: i think no need to label services.

**Review by ddrechse — APPROVED** [review]

**Pull request merged**